util/platform/netdev/NetDev.c

```diff
@@ -20,37 +20,52 @@
 #include "util/platform/netdev/NetDev.h"
 #include "util/platform/netdev/NetPlatform.h"
 
-void NetDev_addAddress(const char* ifName,
-                       struct Sockaddr* sa,
-                       int prefixLen,
-                       struct Log* logger,
-                       struct Except* eh)
-{
-    int addrFam = Sockaddr_getFamily(sa);
+struct AddrBundle {
+    int addrFam;
+    char* printedAddr;
+    void* addr;
+};
 
-    struct Allocator* alloc;
-    BufferAllocator_STACK(alloc, 4096);
-    char* printedAddr = Sockaddr_print(sa, alloc);
-    if (addrFam != Sockaddr_AF_INET && addrFam != Sockaddr_AF_INET6) {
-        Except_throw(eh, "Unknown address type for address [%s]", printedAddr);
+static struct AddrBundle* checkAddressAndPrefix(struct Sockaddr* sa,
+                                  int prefixLen,
+                                  struct Allocator* alloc,
+                                  struct Except* eh)
+{
+    struct AddrBundle* bundle = Allocator_malloc(alloc, sizeof(struct AddrBundle));
```

Comment:
Better to use Allocator_calloc() here because supposing `Sockaddr_getAddress()` screws up you'd have a "wild pointer", it's much harder for me to review code when it doesn't use calloc()...